### PR TITLE
Use Target variable in exists simplifier

### PR DESCRIPTION
**Pull request by ana-pantilie**
Fixes https://github.com/kframework/kore/issues/1551

---

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/src/Kore/Variables/Target.hs [hunk 1]
```diff
@@ -158,3 +159,13 @@ instance
     unparse (NonTarget var) = unparse var
     unparse2 (Target var) = unparse2 var
     unparse2 (NonTarget var) = unparse2 var
+
+targetIfEqual
+    :: Eq variable
+    => variable
```
Comment (ttuegel) [inline kore/src/Kore/Variables/Target.hs]: I think this is only used on `ElementVariable`s anyway, and we should always try to keep the `ElementVariable` or `SetVariable` wrapper:

```suggestion
    => ElementVariable variable
```

File: kore/src/Kore/Step/Simplification/TermLike.hs [hunk 2]
```diff
@@ -389,11 +397,39 @@ simplifyInternal term sideCondition = do
...
                 Equals.simplify sideCondition =<< simplifyChildren equalsF
-            ExistsF exists ->
-                Exists.simplify sideCondition
-                    =<< simplifyChildren (refresh exists)
+            ExistsF exists -> do
+                simplifiedChildren <-
+                    simplifyChildren (refresh exists)
+                targetedResults <-
+                    Exists.simplify
+                        (targetSideCondition sideCondition)
```
Comment (ttuegel) [inline kore/src/Kore/Step/Simplification/TermLike.hs]: Why don't we target the children before simplification? That is, add the `Target` and `NonTarget` wrappers in `refresh. That should eliminate the need to do any re-ordering at all in the `\exists` simplifier.

Comment (ana-pantilie) [reply kore/src/Kore/Step/Simplification/TermLike.hs]: I tried doing that at first but I remember it resulted in some type issues, but maybe I did something wrong along the way and that's why it didn't work. I tried a few solutions and the current one seemed the cleanest, but I can revisit what you suggest if you think it's better.

Comment (ttuegel) [reply kore/src/Kore/Step/Simplification/TermLike.hs]: > I remember it resulted in some type issues

That was expected, but let's merge what you have, and come back to it.

**Pull request merged**